servers/src/common/adapters.rs

```diff
@@ -141,15 +141,14 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 				} else {
 					debug!(
 						LOGGER,
-						"adapter: block invalid after hydration, requesting full block"
+						"adapter: block invalid after hydration, ignoring it"
 					);
-					self.request_block(&cb.header, &addr);
```

Comment:
I don't think we can simply ignore these.
We need some conditional way of knowing if we are syncing or not.
* If we are syncing then we _do not_ want to handle propagated compact blocks and we can just ignore the compact block (this can be earlier in the processing, not just when we try and hydrate it)
* If we are not syncing then we _do_ want to handle the propagated compact block and we _do_ want to go request the full block (otherwise we will run the risk of never seeing the latest propagated block and getting out of sync)

Reply:
Can you guide me where or how I can distinguish these two states?

Reply:
Sorry for taking so much time, the adapter has a sync state struct that you can directly check the current syncing status on.

Reply:
Thanks, I will look into it and will update the PR.